### ISSUE TITLE
-Wtautological-overlap-compare in drivers/infiniband/hw/hfi1/msix.c

### ISSUE THREAD

**Issue by nathanchance** [ClangBuiltLinux/linux#841]
```
drivers/infiniband/hw/hfi1/msix.c:136:22: warning: overlapping comparisons always evaluate to false [-Wtautological-overlap-compare]
        if (type < IRQ_SDMA && type >= IRQ_OTHER)
            ~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~
1 warning generated.
```

Patch sent: https://lore.kernel.org/lkml/20200116222658.5285-1-natechancellor@gmail.com/

**Comment by nathanchance** [comment]
Picked up, I'll close this when it appears in -next: https://lore.kernel.org/lkml/20200125193652.GA30707@jggl/

**Comment by nathanchance** [comment]
Fixed in next-20200128: https://git.kernel.org/next/linux-next/c/79ba4f9310673a8870fda5a062a558923fa65c93
